privacyidea/models.py

```diff
@@ -2127,7 +2132,21 @@ def option_dict(self):
         """
         res = {}
         for option in self.options:
-            res[option.Key] = option.Value
+            if option.Type == "option" or option.Type is None:
```

Comment:
I tested this with SQLlite. An old gateway config does not get read, if the option does not contain a *type* in the database. Different database dialects can return different "empty" values, could be empty strings, or Nones or Nulls.
This breaks with SQLite and probably other databases.

It might make more sense to change this to 
~~~~python
if options.Type == "option" or not option.Type:
~~~~


~~~~

Reply:
ok